src/types/types.ts

```diff
@@ -0,0 +1,12 @@
+import { Dispatch } from '@sumup/collector/build/types';
+
+export interface TrackingProps {
+  /**
+   * Set enable to activate tracking mechanism. Default: false.
+   */
+  enableTracking: boolean;
+  /**
+   * Dispatch
+   */
+  trackingPayload: Dispatch;
+}
```

Comment:
These props should be optional.

```suggestion
export interface TrackingProps {
  /**
   * Set enable to activate tracking mechanism. Default: false.
   */
  enableTracking?: boolean;
  /**
   * Dispatch
   */
  trackingPayload?: Dispatch;
}
```

Reply:
I suppose the Partial extends in the component interface should have the same constrain for types.
https://www.typescriptlang.org/docs/handbook/utility-types.html#partialt

Reply:
What's the benefit of making them required here and then using `Partial` in each component?

Reply:
Hmm true. I was thinking that ideally, we should have a standard tracking interface and let each component decide what make sense to them. But you're right. The way I do it now only make it confused and conflict itself. Will change it. Thanks!!